portfolio/src/main/webapp/script.js

```diff
@@ -13,60 +13,60 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * Add typing animation effect
  */
-var TxtRotate = function(el, toRotate, period) {
-  this.toRotate = toRotate;
-  this.el = el;
-  this.loopNum = 0;
-  this.period = parseInt(period, 10) || 2000;
-  this.txt = '';
-  this.tick();
-  this.isDeleting = false;
-};
+class TxtRotate {
+    constructor(el, toRotate, period) {
+    this.toRotate = toRotate;
+    this.el = el;
+    this.loopNum = 0;
+    this.period = parseInt(period, 10) || 2000;
+    this.txt = '';
+    this.isDeleting = false;
+    this.delta =300 - Math.random() * 100;
+    }
 
-TxtRotate.prototype.tick = function() {
-  let i = this.loopNum % this.toRotate.length;
-  let fullTxt = this.toRotate[i];
+    tick() {
+        let i = this.loopNum % this.toRotate.length;
+        let fullTxt = this.toRotate[i];
 
-  if (this.isDeleting) {
-    this.txt = fullTxt.substring(0, this.txt.length - 1);
-  } else {
-    this.txt = fullTxt.substring(0, this.txt.length + 1);
-  }
+        if (this.isDeleting) {
+            this.txt = fullTxt.substring(0, this.txt.length - 1);
+        } else {
+            this.txt = fullTxt.substring(0, this.txt.length + 1);
+        }
 
-  this.el.innerHTML = '<span class="wrap">'+this.txt+'</span>';
+        this.el.innerHTML = '<span class="wrap">'+this.txt+'</span>';
 
-  let that = this;
-  let delta = 300 - Math.random() * 100;
+        let delta = 300 - Math.random() * 100;
 
-  if (this.isDeleting) { delta /= 2; }
+        if (this.isDeleting) { 
+            delta /= 2; 
+            }
 
-  if (!this.isDeleting && this.txt === fullTxt) {
-    delta = this.period;
-    this.isDeleting = true;
-  } else if (this.isDeleting && this.txt === '') {
-    this.isDeleting = false;
-    this.loopNum++;
-    delta = 1000;
-  }
+        if (!this.isDeleting && this.txt === fullTxt) {
+            delta = this.period;
+            this.isDeleting = true;
+        } else if (this.isDeleting && this.txt === '') {
+            this.isDeleting = false;
+            this.loopNum++;
+            delta = 1000;
+        }
 
-  setTimeout(function() {
-    that.tick();
-  }, delta);
-};
+        setTimeout(this.tick.bind(this), delta);
```

Comment:
I think you can use
```
() => {
  this.tick();
}
```
in place of `this.tick.bind(this)`, and this is the preferred way.